pkg/reconciler/githubsource/webhook_client.go

```diff
@@ -36,16 +37,26 @@ type webhookOptions struct {
 }
 
 type webhookClient interface {
-	Create(ctx context.Context, options *webhookOptions) (string, error)
-	Delete(ctx context.Context, options *webhookOptions, hookID string) error
+	Create(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error)
+	Delete(ctx context.Context, options *webhookOptions, hookID, alternateGitHubAPIURL string) error
 }
 
 type gitHubWebhookClient struct{}
 
-func (client gitHubWebhookClient) Create(ctx context.Context, options *webhookOptions) (string, error) {
+func (client gitHubWebhookClient) Create(ctx context.Context, options *webhookOptions, alternateGitHubAPIURL string) (string, error) {
 	logger := logging.FromContext(ctx)
 
 	ghClient := client.createGitHubClient(ctx, options)
+	if alternateGitHubAPIURL != "" {
+		//This is to support GitHub Enterprise, this might be something like https://github.company.com/api/v3/
+		baseURL, err := url.Parse(alternateGitHubAPIURL)
+		if err != nil {
+			logger.Infof("failed to create webhook - error occured parsing githubAPIURL %s, error was %v", alternateGitHubAPIURL, err)
```

Comment:
Prefer `%q` for the string here; it will quote the URL and will help with finding e.g. leading or trailing spaces.

Reply:
Modified in latest commit